docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,119 @@
+            hive.openshift.io/aro-snowflake: "true"
+          matchExpressions:
+            - key: hive.openshift.io/limited-support
+              operator: NotIn
+              values: 
+                - "true"
+            - key: hive.openshift.io/limited-support
+              operator: NotExists
+        additionalClusterDeploymentLabels:
+          prom_label_name: hive.openshift.io/cd-label-key
+```
+### clusterDeploymentRelatedMetrics
+We are only considering with metrics that are calculated whenever we parse the cluster deployment, or is tied to a cluster deployment, so we will group them all under hiveConfig.spec.metricsConfig.clusterDeploymentRelatedMetrics. Each metric that needs to be customized would need a separate entry.
+Implementation would require a separate class that would encapsulate all the clusterDeploymentRelatedMetrics, and related methods for registering and observing the metrics.
+
+#### minimumDuration
+Deprecate the current implementation of hiveConfig.spec.metricsConfig.metricsWithDuration, and change it to be reported as metricsConfig.clusterDeploymentRelatedMetrics[$name].minimumDuration. The implementation of using the duration as a threshold before we report the metric stays the same.
+
+#### additionalClusterDeploymentLabels
+Deprecate current implementation of hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels and change it to be reported as metricsConfig.clusterDeploymentRelatedMetrics[$name].additionalClusterDeploymentLabels. It's implementation will not change.
+
+#### clusterDeploymentLabelSelector
+This would be a new feature, of type [LabelSelector](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector), and we'd use the LabelSelector.MatchLabels and/or LabelSelector.MatchExpressions to match the conditions in order to decide if a metric should be reported.
+This encapsulates slightly more advanced filter logic over the existing clusterdeployment labels. In the example above, hive_foo_counter metric will only be reported for the cluster deployments that are labelled with aro-snowflake, not in limited support, and has the metric value > 10m.
+
+### Implementation Details / Notes
+- In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.`
+- Current configuration of using a shorthand camelCase key to refer to a corresponding metric while configuring hiveconfig.spec.metricsConfig.metricsWithDuration will be removed, and users will have to use the full metric name in the proposed configuration.
+- Additionally, we would also want to remove `cluster_type` as a fixed label for all the metrics that do report it now. This would be a breaking change as the consumers who want `cluster_type` will have to configure it via metricsConfig.clusterDeploymentRelatedMetrics[].additionalClusterDeploymentLabels for all the metrics they need it for.
```

Comment:
Just to clarify: We will be able to detect whether the old or new config mechanism is in play (esp if we make them mutually exclusive as suggested above), so it wouldn't be crazy to leave the `cluster_type` fixed label for the old way, but remove it for the new, right?